### DISCUSSION TITLE
[Serializer] Can't reproduce DeserializeNestedArrayOfObjectsTest

### DISCUSSION THREAD

**Discussion by Hanmac** [symfony/symfony#52623] [Q&A]
i tried to reproduce this test:
https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/Serializer/Tests/DeserializeNestedArrayOfObjectsTest.php

in Phpsandbox:

https://phpsandbox.io/e/x/xl1r9?layout=EditorPreview&defaultPath=%2F&theme=dark&showExplorer=no&openedFiles=

but i failed.

What exactly I'm missing there?

i know with extra Extractors i could get it working with the `add` method, but that's different

**Comment by ghost** [comment]
Online sandboxes have some functionality turned off or just not configured. I would use local environment to do the same.

**Reply by Hanmac** [reply]
The Fail is that the Animal Entity aren't denormalized

The Test did: `self::assertInstanceOf(Animal::class, $zoo->getAnimals()[0]);`

with another extractor for the `add` function, it works

i just wanted to learn what is wrong with this one

**Reply by ghost** [reply]
I've copy pasted the index.php from your sandbox and it looks fine
```sh
$ php index.php
object(ZooImmutable)#52 (1) {
  ["animals":"ZooImmutable":private]=>
  array(1) {
    [0]=>
    object(Animal)#51 (1) {
      ["name":"Animal":private]=>
      string(3) "Bug"
    }
  }
}
```
I have symfony/serializer:6.2.10 (other dependencies used in your code sample: symfony/property-info:6.2.10, symfony/property-access:6.2.8, phpdocumentor/type-resolver:1.7.1) 
PHP 8.2.5

**Reply by Hanmac** [reply]
This user had the similar problem (in Slack) trying to unserialize:
 https://symfony-devs.slack.com/archives/CEUD3992S/p1700136733671479

what helped for them was the addition of `add` function

**Reply by ghost** [reply]
I can't help you more. As I've shown in the previous comment, it's working fine for me.
